### PR TITLE
Remove function set state from useControlledState

### PR DESCRIPTION
**Pull request by devongovett**
Closes https://github.com/adobe/react-spectrum/issues/2320

### CODE REVIEW

**Review by LFDanLu — DISMISSED** [review]
Tested locally and it seems like the components behave as expected (at least the ones with directly modified stately hooks). Will need to do a dedicated testing session for this change though haha

**Review by LFDanLu — APPROVED** [review]
Re-approving, components still seem to work, type update looks correct

**Review by ktabors — DISMISSED** [review]
I tested controlled state in numberfield, checkbox group, and color field and they all worked.

File: packages/@react-spectrum/numberfield/test/NumberField.test.js [hunk 1]
```diff
@@ -1675,7 +1675,7 @@ describe('NumberField', function () {
     expect(onChangeSpy).toHaveBeenCalledWith(10123);
   });
 
-  it.skip.each`
+  it.each`
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/numberfield/test/NumberField.test.js:1678]: fixed by this PR

File: packages/@react-spectrum/button/stories/ToggleButton.stories.tsx [hunk 2]
```diff
@@ -84,3 +87,14 @@ function render(props = {}) {
     </ToggleButton>
   </Flex>);
 }
+
+function ControlledToggleButton() {
+  let [selected, setSelected] = useState(false);
+  return (
+    <div>
+      <ToggleButton onChange={setSelected}>Press Me</ToggleButton>
```
Comment (ktabors) [inline packages/@react-spectrum/button/stories/ToggleButton.stories.tsx:95]: Could we change this too?
```suggestion
      <ToggleButton isSelected={selected} onChange={setSelected}>Press Me</ToggleButton>
```

**Review by LFDanLu — APPROVED** [review]
LGTM

**Review by LFDanLu — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]
Did well in testing session

**Pull request merged**